### ISSUE TITLE
ec2 security group - TypeError: unhashable type: 'list'

### ISSUE THREAD

**Issue by tekumara** [spulec/moto#4974]
_unhashable.py_:
```python
import boto3
from moto import mock_ec2
from moto.ec2 import ec2_backends
from moto.ec2.models import AMIS

mock = mock_ec2()
mock.start()

ec2_client = boto3.client("ec2", region_name="us-east-1")
ec2_client.run_instances(
    MaxCount=1,
    MinCount=1,
    NetworkInterfaces=[
        {
            "DeviceIndex": 0,
            "Description": "Primary network interface",
            "DeleteOnTermination": True,
            "SubnetId": next(ec2_backends["us-east-1"].get_all_subnets()).id,
            "Ipv6AddressCount": 0,
            "Groups": ["default"],
        }
    ],
)
```

```
$ python unhashable.py    
Traceback (most recent call last):
  File "/Users/tekumara/code/aec/tests/unhashable.py", line 10, in <module>
    ec2_client.run_instances(
  File "/Users/tekumara/code/aec/.venv/lib/python3.9/site-packages/botocore/client.py", line 401, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/Users/tekumara/code/aec/.venv/lib/python3.9/site-packages/botocore/client.py", line 717, in _make_api_call
    http, parsed_response = self._make_request(
  File "/Users/tekumara/code/aec/.venv/lib/python3.9/site-packages/botocore/client.py", line 737, in _make_request
    return self._endpoint.make_request(operation_model, request_dict)
  File "/Users/tekumara/code/aec/.venv/lib/python3.9/site-packages/botocore/endpoint.py", line 107, in make_request
    return self._send_request(request_dict, operation_model)
  File "/Users/tekumara/code/aec/.venv/lib/python3.9/site-packages/botocore/endpoint.py", line 183, in _send_request
    while self._needs_retry(attempts, operation_model, request_dict,
  File "/Users/tekumara/code/aec/.venv/lib/python3.9/site-packages/botocore/endpoint.py", line 305, in _needs_retry
    responses = self._event_emitter.emit(
  File "/Users/tekumara/code/aec/.venv/lib/python3.9/site-packages/botocore/hooks.py", line 358, in emit
    return self._emitter.emit(aliased_event_name, **kwargs)
  File "/Users/tekumara/code/aec/.venv/lib/python3.9/site-packages/botocore/hooks.py", line 229, in emit
    return self._emit(event_name, kwargs)
  File "/Users/tekumara/code/aec/.venv/lib/python3.9/site-packages/botocore/hooks.py", line 212, in _emit
    response = handler(**kwargs)
  File "/Users/tekumara/code/aec/.venv/lib/python3.9/site-packages/botocore/retryhandler.py", line 194, in __call__
    if self._checker(**checker_kwargs):
  File "/Users/tekumara/code/aec/.venv/lib/python3.9/site-packages/botocore/retryhandler.py", line 267, in __call__
    should_retry = self._should_retry(attempt_number, response,
  File "/Users/tekumara/code/aec/.venv/lib/python3.9/site-packages/botocore/retryhandler.py", line 286, in _should_retry
    return self._checker(attempt_number, response, caught_exception)
  File "/Users/tekumara/code/aec/.venv/lib/python3.9/site-packages/botocore/retryhandler.py", line 333, in __call__
    checker_response = checker(attempt_number, response,
  File "/Users/tekumara/code/aec/.venv/lib/python3.9/site-packages/botocore/retryhandler.py", line 233, in __call__
    return self._check_caught_exception(
  File "/Users/tekumara/code/aec/.venv/lib/python3.9/site-packages/botocore/retryhandler.py", line 376, in _check_caught_exception
    raise caught_exception
  File "/Users/tekumara/code/aec/.venv/lib/python3.9/site-packages/botocore/endpoint.py", line 246, in _do_get_response
    responses = self._event_emitter.emit(event_name, request=request)
  File "/Users/tekumara/code/aec/.venv/lib/python3.9/site-packages/botocore/hooks.py", line 358, in emit
    return self._emitter.emit(aliased_event_name, **kwargs)
  File "/Users/tekumara/code/aec/.venv/lib/python3.9/site-packages/botocore/hooks.py", line 229, in emit
    return self._emit(event_name, kwargs)
  File "/Users/tekumara/code/aec/.venv/lib/python3.9/site-packages/botocore/hooks.py", line 212, in _emit
    response = handler(**kwargs)
  File "/Users/tekumara/code/aec/.venv/lib/python3.9/site-packages/moto/core/models.py", line 288, in __call__
    status, headers, body = response_callback(
  File "/Users/tekumara/code/aec/.venv/lib/python3.9/site-packages/moto/core/responses.py", line 205, in dispatch
    return cls()._dispatch(*args, **kwargs)
  File "/Users/tekumara/code/aec/.venv/lib/python3.9/site-packages/moto/core/responses.py", line 315, in _dispatch
    return self.call_action()
  File "/Users/tekumara/code/aec/.venv/lib/python3.9/site-packages/moto/core/responses.py", line 405, in call_action
    response = method()
  File "/Users/tekumara/code/aec/.venv/lib/python3.9/site-packages/moto/ec2/responses/instances.py", line 82, in run_instances
    new_reservation = self.ec2_backend.add_instances(
  File "/Users/tekumara/code/aec/.venv/lib/python3.9/site-packages/moto/ec2/models.py", line 1143, in add_instances
    new_instance = Instance(
  File "/Users/tekumara/code/aec/.venv/lib/python3.9/site-packages/moto/ec2/models.py", line 741, in __init__
    self.prep_nics(
  File "/Users/tekumara/code/aec/.venv/lib/python3.9/site-packages/moto/ec2/models.py", line 1042, in prep_nics
    use_nic = self.ec2_backend.create_network_interface(
  File "/Users/tekumara/code/aec/.venv/lib/python3.9/site-packages/moto/ec2/models.py", line 498, in create_network_interface
    eni = NetworkInterface(
  File "/Users/tekumara/code/aec/.venv/lib/python3.9/site-packages/moto/ec2/models.py", line 360, in __init__
    self.ec2_backend.groups[subnet.vpc_id][group_id] = group
TypeError: unhashable type: 'list'
```

Present in moto 3.1.2 but not previous versions.

**Comment by bblommers** [comment]
Hi @tekumara, thanks for raising this! We refactored our EC2 backend a bit, but we didn't have a test for this specific scenario, so we didn't know that was broken. Will push a fix shortly.

**Comment by bblommers** [comment]
This should be fixed with Moto 3.1.3 that was just released. Please let us know if you're running into any other issues!
